builtin.c

```diff
@@ -0,0 +1,104 @@
+/*    builtin.c
+ *
+ *    Copyright (C) 2021 by Paul Evans and others
+ *
+ *    You may distribute under the terms of either the GNU General Public
+ *    License or the Artistic License, as specified in the README file.
+ *
+ */
+
+/* This file contains the code that implements functions in perl's "builtin::"
+ * namespace
+ */
+
+#include "EXTERN.h"
+#include "perl.h"
+
+#include "XSUB.h"
+
+XS(XS_builtin_true);
+XS(XS_builtin_true)
+{
+    dXSARGS;
+    if(items)
+        croak_xs_usage(cv, "");
+    XSRETURN_YES;
+}
+
+XS(XS_builtin_false);
+XS(XS_builtin_false)
+{
+    dXSARGS;
+    if(items)
+        croak_xs_usage(cv, "");
+    XSRETURN_NO;
+}
+
+XS(XS_builtin_isbool);
+XS(XS_builtin_isbool)
+{
+    dXSARGS;
+    if(items != 1)
+        croak_xs_usage(cv, "sv");
+
+    SV *sv = ST(0);
+    if(SvIsBOOL(sv))
+        XSRETURN_YES;
+    else
+        XSRETURN_NO;
+}
+
+XS(XS_builtin_import);
+XS(XS_builtin_import)
+{
+    dXSARGS;
+    int i;
```

Comment:
You can declare `i` inside `for()`.

Reply:
Ahyes, I forget that we're now in the lovely modern(!) world of C99. Such heady heights...

Reply:
Done 1a1f12ffb0